### ISSUE TITLE
Very slow step time on FDS 6.7.8 versus FDS 6.7.7

### ISSUE THREAD

**Issue by jstebbi** [firemodels/fds#10660]
I am trying to run FDS 6.7.8 on Windows 10 and I am getting very slow step times for a simulation that previously ran perfectly on 6.7.7.  

I tried on several different Windows 10 computers with different mesh types (mesh using MULT, 8, 16, 18 meshes, mesh line by line assigned to cores) with the same results.  The case doesn't even reach .19s simulation time until the 1000 time step, and is very slow after that, effectively it will never finish.  I even tried to run on an old Linux computer and got identical results.

The attached code runs in approximately ~7 hours on FDS 6.7.7 on my current Windows setup.  I run it with 18 meshes on 18 MPI processes with OpenMP disabled.  I attempted on 6.7.8 with the MPI only .exe file.

I have attached a simplified .fds file that runs into the same issue when attempting to run on 6.7.8 on my system.

The OBST lines are generated with BlenderFDS v5.1.0.

Thanks in advance!
[Candle7_21.fds.txt](https://github.com/firemodels/fds/files/8854003/Candle7_21.fds.txt)

**Comment by mcgratta** [comment]
I'll run the case. Is the time step significantly different in the two runs? Look in the _steps.csv files.

**Comment by mcgratta** [comment]
There is definitely a much different time step. I'll try to figure out why.

**Comment by jstebbi** [comment]
Thank you.  Maybe it might be the voxel created OBST in Blender.  It's probably the one thing in there that wasn't tested.

FYI:

FDS 6.7.7 Case:

Time Step	Wall Time	Step Size	Simulation Time	CPU Time
1	2022-05-24T11:20:16.214-04:00	1.83E-02	0.0183	2.81E+00

600	2022-05-24T11:20:21.150-04:00	2.83E-03	2.05288	7.67E+00
700	2022-05-24T11:20:21.943-04:00	2.54E-03	2.33048	8.45E+00
800	2022-05-24T11:20:22.728-04:00	2.54E-03	2.58474	9.23E+00
900	2022-05-24T11:20:23.523-04:00	2.54E-03	2.83901	1.00E+01
1000	2022-05-24T11:20:24.302-04:00	2.54E-03	3.09328	1.08E+01


FDS 6.7.8 Case:

Time Step	Wall Time	Step Size	Simulation Time	CPU Time
1	2022-06-07T11:55:55.066-04:00	1.83E-02	0.0183	2.75E+00

600	2022-06-07T11:55:58.441-04:00	1.21E-04	0.14387	6.05E+00
700	2022-06-07T11:55:58.988-04:00	1.21E-04	0.15602	6.59E+00
800	2022-06-07T11:55:59.488-04:00	1.21E-04	0.16816	7.09E+00
900	2022-06-07T11:55:59.972-04:00	1.21E-04	0.1803	7.58E+00
1000	2022-06-07T11:56:00.488-04:00	1.21E-04	0.19244	8.09E+00


If you extrapolate a bit once it gets to 10000 or so time step, you can see that the 6.7.7 case ends in about 7 hours and the 6.7.8 case takes about 5 days.

**Comment by mcgratta** [comment]
Yes, the time step is significantly different. I am bisecting now to pin down the exact code change that causes the problem.

**Comment by mcgratta** [comment]
I found the problem. In February a change was made to make the time step more restrictive when using HT3D. I'm going to reassign this issue to Randy McDermott because he knows more about why that change was made. For the moment, you will have to stay with the older version because I do not have an easy work-around for this.

```
c9c1d2270b0fa4d59d98e4387021fef2daa23934 is the first bad commit
commit c9c1d2270b0fa4d59d98e4387021fef2daa23934
Author: rmcdermo <randy.mcdermott@gmail.com>
Date:   Tue Feb 1 11:11:20 2022 -0500

    FDS Source: make CHECK_HT=T default for HT3D

:040000 040000 d0b5db02d6433ba0dd49dd1043097b6ac297c47f e6f1e61c34d77bcab16ba2ef6c5ffc859ee89898 M      Source
```

**Comment by jstebbi** [comment]
Will keep using 6.7.7 for now.  Thank you.

**Comment by rmcdermo** [comment]
I am not sure what the issue is here.  If you run the case with CHECK_HT=F and the latest code does that work?

**Comment by jstebbi** [comment]
I am running a couple simulations right now on 6.7.7.  I will install 6.7.8 and try this first thing on Monday and let you know.

**Comment by mcgratta** [comment]
Install the latest version, which is 6.7.9.

**Comment by jstebbi** [comment]
Will do.

**Comment by jstebbi** [comment]
I tried to run the code I posted in 6.7.9. with the only change being adding the MISC line CHECK_HT=F.  The code ran similarly slow, this time advancing the Step Time faster but the Simulation Time very slow, as before.

Time Step	Wall Time	                        Step Size	Simulation Time	CPU Time
500	       2022-07-25T08:53:59.569-04:00	1.06E-04	 0.11026	                4.98E+00
600	       2022-07-25T08:54:00.116-04:00	9.57E-05	 0.11989	                5.53E+00
700	       2022-07-25T08:54:00.647-04:00	8.61E-05	 0.1291	                6.06E+00
800	       2022-07-25T08:54:01.194-04:00	8.61E-05	 0.13771	                6.61E+00
900	        2022-07-25T08:54:01.724-04:00	8.61E-05	 0.14632	                7.14E+00
1000	2022-07-25T08:54:02.255-04:00	8.61E-05	 0.15493	                7.67E+00

**Comment by mcgratta** [comment]
Our mistake. I just checked the code and at present, you cannot disable CHECK_HT if you are using HT3D. So you have two options:
1. Do not use HT3D.
2. Wait for me to fix this problem and post a new version.

**Comment by jstebbi** [comment]
Thanks for the feedback.

I was advised a while back that for the type of simulation I am doing that HT3D is the best way to get wall temperatures that have accuracy.  So, I will continue using 6.7.7. for now.  Will keep checking for the new versions.  Thanks.

**Comment by mcgratta** [comment]
Advised by whom? HT3D is not as accurate as the default 1-D heat conduction routine in solids. The only advantage of HT3D is the inclusion of 3-D heat transfer, but that may not be important here.

**Comment by jstebbi** [comment]
dr_jfloyd said (https://groups.google.com/g/fds-smv/c/Peno2yFaDgw/m/5g4c-YhHAgAJ) for my candle case, "If accurate temperatures of the rim is the critical output, you probably need to think about using HT3D for the container and having a much finer grid resolution."

I am attempting to use FDS to model a candle rim temperature and sidewall temperature  for a small poured candle with a 35mm width and 2.5mm thick walled glass container.

**Comment by drjfloyd** [comment]
My post said if the wall temperature is critical to consider HT3D AND a much finer grid resolution. Comparing the input file in this issue to the file you posted in the thread you mentioned, you have only added HT3D, your grid resolution is still ~5 mm.  The 1D heat transfer solver selects node sizes to try and ensure reasonably accurate wall temperature profile. With the HT3D solver, your node resolution is the same as the grid size. It is up to you to ensure that the grid size is small enough to obtain reasonable results from the 3D heat transfer solver. The 1D solver would use cells about an order of magnitude smaller than your current mesh.

**Comment by jstebbi** [comment]
Thanks for your response.  In my current iterations I am using 2.5mm and 2mm cell size for the bigger and smaller size candles, respectively.  Do you think it would make sense to use these with HT3D?

Part of my study is for candle temperatures at 2-4 hours, so a 1mm or .5mm cell size would lead to exceedingly long simulation time.  However, if you think it is absolutely necessary to go even smaller than 2.5mm I can try and create a model to predict 4 hour temperature based on a much shorter simulation with 1mm or lower cell size.

**Comment by drjfloyd** [comment]
If the wall temperature is the critical quantity, you need do a grid study to evaluate this. You have BNDF files with fluxes from the flame to the candle rim. You could create a simpler test case of just a portion of the rim with mirror conditions as appropriate, apply the average flux from the flame, and experiment with 1D and 3D heat transfer and the impact of grid resolution.

**Comment by jstebbi** [comment]
I will try and do a study with a lower resolution on a shorter simulation time.  Maybe I can try 1.25mm and .5mm so I can keep the candle OBST dimensions the same.  The actual candle being modeled has a ~2.5mm rim width so that's why I chose that grid size.

I will try this this week.  Thanks for the help.

**Comment by jstebbi** [comment]
Would you suggest changing the voxel size on the OBST to match the grid size?  Or can I leave it at 2.5mm and just refine the grid itself?

**Comment by drjfloyd** [comment]
With 3D heat transfer, how well do you think you will be resolving heat transfer across the rim and down into the body of the candle when you only have one node temperature in the rim?

**Comment by jstebbi** [comment]
I actually have one node on the opposite side and two nodes on the sidewalls just below those.  Do you think I need a couple more?

I found the temperatures on the sidewalls were very close to the real life burn tests when the temperature of the rim matched that of the rim temperature measured in the burn test for the same simulation, so I thought that was promising.

**Comment by jstebbi** [comment]
If I take the rim temperature at 4 places 90 degrees from each other and then averaged them, that might give me a better resolution of real world testing.

**Comment by jstebbi** [comment]
Hi all,

I performed some grid study and 1D tests.

In FDS 6.7.7., It appears that when I lower the grid size from 2.5mm to 1.25mm on an identical system using HT3D in a 600 second simulation, the Wall Temperatures averaged over the last 60 seconds are:

**2.50mm mesh**
53.757 Highest Average Outer rim Wall Temp
33.419 Average Sidewall Wall Temp

**1.25 mesh**
54.950 Highest Average Outer rim Wall Temp
33.622 Average Sidewall Wall Temp

![image](https://user-images.githubusercontent.com/93282682/183755840-0df54585-26fd-4903-9fce-5d280a80ec49.png)

![image](https://user-images.githubusercontent.com/93282682/183755182-02e9df9b-ef7f-4951-8af5-8abb91af07b7.png)


The graphs were similar as far as ramp and fluctuation.

---

_I then used FDS 6.7.9. to do a 8000s 1D and FDS 6.7.7. 8000s HT3D comparison._

**FDS 677 with HT3D**
63.380 Highest Average Outer Rim Wall Temp
51.092 Average Sidewall Wall Temp

**FDS 6.7.9. with 1D**
62.762 Highest Average Outer rim Wall Temp
48.961 Average Sidewall Wall Temp

![image](https://user-images.githubusercontent.com/93282682/183757756-fa9e855e-d4d9-470a-be07-4af6409497ff.png)

![image](https://user-images.githubusercontent.com/93282682/183758079-edc83127-ce1e-4ff5-b123-4503ef2edbab.png)


I also added two more Wall Temperature Devc for a total of 4 Rim Temperature thermocouples and I found that one of the Wall Temps I was using up to now was the highest temperature at the end of the simulation.

The 1D simulation seemed to peak Temp very quickly and then fluctuate throughout.  Also, should be noted that 1D in 6.7.9 took ~26 hours and HT3D in 6.7.7 ran for ~50 hours.
